doc/BUILD.rst

```diff
@@ -0,0 +1,722 @@
+
+You can also have the ``setup`` command contain certain build options (seen
+below). The following are an *example* and *not required* for getting Taisei
+building.
+
+.. code:: sh
+
+   # enables Developer Mode and debugging symbols
+   meson setup -C build/ -Ddeveloper=true -Dbuildtype=debug
+
+You can then apply more build options later using ``meson configure`` (as seen
+below). It will automatically reconfigure your build environment with the new
+options without having to rebuild everything.
+
+System/Vendored Dependencies (``--wrap-mode``)
+""""""""""""""""""""""""""""""""""""""""""""""
+
+See: `Meson Manual <https://mesonbuild.com/Wrap-dependency-system-manual.html>`__
+
+* Default: ``default``
+* Options: ``default``, ``nofallback``, ``forcefallback``, ...
+
+This is a core ``meson`` flag that does quite a few things. Not all of them will
+be covered here. Refer to the ``meson`` documentation linked above.
+
+Generally, ``default`` will rely on system-installed libraries when available,
+and fallback to vendored in-repository dependencies when necessary.
+
+``forcefallback`` will heavily encourage the use of in-repository dependencies
+whenever possible. Recommended for release builds.
+
+``nofallback`` discourages the use of in-repository dependencies whenever
+possible, instead relying on system libraries. Useful for CI.
+
+.. code:: sh
+
+   # for release builds
+   meson configure build/ --wrap-mode=forcefallback
+   # useful for testing/CI
+   meson configure build/ --wrap-mode=nofallback
+
+Relative Directory Install (``-Dinstall_relative``)
+"""""""""""""""""""""""""""""""""""""""""""""""""""
+
+* Default: ``true`` or ``false`` (platform-dependent)
+
+``-Dinstall_relative`` is a special option that changes depending on the
+platform build target.
+
+It is set to ``true`` when building for macOS, Windows, Emscripten, and Switch.
+
+It is set to ``false`` when building for Linux.
+
+Install Prefix (``--prefix``)
+"""""""""""""""""""""""""""""
+
+* Default: ``/usr/local``
+
+``--prefix`` installs the Taisei binary and content files to a path of your
+choice on your filesystem.
+
+On Linux without ``-Dinstall_relative`` enabled (i.e: ``false``), it should be
+set to ``/usr/local``.
+
+On other platforms, it will install all Taisei game files to the directory of
+your choice.
+
+.. code:: sh
+
+   meson setup --prefix=/path/goes/here -C build/
+
+Package Data (``-Dpackage_data``)
+"""""""""""""""""""""""""""""""""
+
+* Default: ``auto``
+* Options: ``auto``, ``true``, ``false``
+
+Packages data into either a glob or a ``.zip`` depending on if ``-Denable_zip``
+is ``true`` (see below).
+
+.. code:: sh
+
+   meson configure build/ -Dpackage_data=false
+
+Enable ZIP Loading (``-Denable_zip``)
+"""""""""""""""""""""""""""""""""""""
+
+* Default: ``true```
+* Options: ``true``, ``false``
+
+Controls whether or not Taisei can load game data (textures, shaders, etc) from
+``.zip`` files. Useful for distribution and packaging.
```

Comment:
Worth mentioning that setting this option to `false` also disables `package_data` by default.